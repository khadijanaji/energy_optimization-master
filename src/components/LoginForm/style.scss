.login {
    text-align: center;
    background-color: #f8f8f8;
    height: 100vh;
    &__header {
        background-image:linear-gradient(308deg, #485563, #29323c);    
        width: 100%;
        padding-bottom: 40px;
        &--logo {
            height: 80px;
            margin-top: 60px;
        }
        &--text1 {
            color: #ffffff;
            font-size: 24px;
            text-shadow: 0 2px 2px #29323c;
            margin-top: 45px;
        }
    }

    &__text2 {
        font-size: 14px;
        width: 243px;
        font-family: Roboto;
        margin-top: 42px;
        margin-left: 50%;
        transform: translate(-50%,-50%);
    }

    &__email {
        margin-top: 13%;
        width: 90%;
        margin-left: 50%;
        transform: translate(-50%,-50%);
    }
    
    &__password {
        width: 90%;
        margin-top: 11%;
        margin-left: 50%;
        transform: translate(-50%,-50%);
    }
    &__button {
        width: 90%;
        margin-left: 50%;
        transform: translate(-50%,-50%);
        height: 50px; 
        margin-top: 10%;
        & > button{
            background-color: rgb(117, 166, 21);
            color: rgb(255, 255, 255);
            font-weight: bold;
            border-radius: 4px;
            height: 56px;
            width: 100%;
        }
        & > button:disabled{
            background-color: #ccc;
        }
        & > button:hover{
            background-color: rgb(147, 196, 51);
        }
    }
    &__error {
        width: 90%;
        transform: translate(-50%, -50%);
        margin-left: 50%;
        text-align: left;
        padding-bottom: 19px;
        &--text {
            color: #ea3d27;
            font-size: 13px;
            text-align: left;
            display: inline-block;
            width: 95%;
        }
        &--icon {
            display: inline-block;
            width: 4%;
            text-align: right;
        }
    }
   
}