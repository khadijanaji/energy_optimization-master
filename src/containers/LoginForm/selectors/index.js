// @flow

import {getFormValues, isInvalid, isPristine} from "redux-form";

import type {LoginFormStoreRecordType} from "../../../storeTypes/LoginForm";

const loginFormName = "loginForm";

const globalLoginFormSelector = (state: any): LoginFormStoreRecordType => state.login;

const loginFormSelector = (state: any): any => ({
  ...globalLoginFormSelector(state).toJS(),
  "isPristine": Boolean(isPristine(loginFormName)(state)),
  "isInvalid": Boolean(isInvalid(loginFormName)(state))
});

const emailPasswordSelector = (state: any) => getFormValues(loginFormName)(state);

const tokenSelector = (state: any): string => {
  const {"infoUser": {token}} = globalLoginFormSelector(state) || {};
  return token;
};

const emailNameSelector = (state: any) => {
  const {"infoUser": {email, name, roles, factories}} = globalLoginFormSelector(state) || {};
  return {
    email,
    name,
    roles,
    factories
  };
};

export {
  loginFormName,
  globalLoginFormSelector,
  emailPasswordSelector,
  tokenSelector,
  emailNameSelector,
  loginFormSelector
};
