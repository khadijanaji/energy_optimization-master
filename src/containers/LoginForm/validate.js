// @flow

const validate = (values: any) => {
  const errors = {};
  if (values && !values.email) {
    errors.email = "Ce champs est obligatoire!";
  }
  if (values && !values.password) {
    errors.password = "Ce champs est obligatoire!";
  }
  return errors;
};

export default validate;
