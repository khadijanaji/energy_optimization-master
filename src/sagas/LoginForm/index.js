// @flow
import {takeLatest} from "redux-saga/effects";

import {LOGIN_ACTION} from "../../actions/loginActions/constants";
import login from "./login";

/**
 * Listens for the get item groups action and trigger the request flow, cancelling the previous
 * request if it was still ongoing.
 * @return {null} nothing
 */
export default function *(): Generator<any, void, any> {
  yield takeLatest(LOGIN_ACTION, login);
}
