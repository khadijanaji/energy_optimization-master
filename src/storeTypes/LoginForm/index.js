// @flow

import type {RecordFactory, RecordOf} from "immutable";
import {Record} from "immutable";

export type InfoUserPropsType = {
    "email": string,
    "name": string,
    "token": string,
    "roles": Array<string>,
    "factories": Array<string>
};

export type LoginFormStoreType = {
    "status": "Idle" | "Loading" | "Success" | "Error",
    "onSubmit": Function,
    "errorMsg"?: string,
    "isPristine": boolean,
    "isInvalid": boolean,
    "infoUser": InfoUserPropsType
};

export type InfoUserStoreRecordFactoryType = RecordFactory<InfoUserPropsType>;

export const InfoUserStoreRecordFactory: InfoUserStoreRecordFactoryType = Record({
  "email": "",
  "name": "",
  "token": "",
  "roles": [],
  "factories": []
}, "InfoUserStoreRecordFactory");

export type LoginFormStoreRecordType = RecordOf<LoginFormStoreType>;

export type LoginFormStoreRecordFactoryType = RecordFactory<LoginFormStoreType>;

export const LoginFormStoreRecordFactory: LoginFormStoreRecordFactoryType = Record({
  "status": "Idle",
  "onSubmit": () => ({}),
  "errorMsg": "",
  "isPristine": true,
  "isInvalid": true,
  "infoUser": {
    "email": "",
    "name": "",
    "token": "",
    "roles": [],
    "factories": []
  }
}, "LoginFormStoreRecordFactory");
