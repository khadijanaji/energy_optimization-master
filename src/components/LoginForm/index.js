/* eslint-disable react/no-multi-comp */
/* eslint-disable react/no-set-state */
// @flow

import React from "react";
import FormControl from "@material-ui/core/FormControl";
import {Field} from "redux-form";

import TextField from "../common/reduxFormFields/TextField";
import Button from "../common/reduxFormFields/Button";
import logo from "../../img/reporting-192.png";
import warn from "../../img/login/text-warning-icon.svg";
import {RemoveRedEye} from "@material-ui/icons";
import {InputAdornment} from "@material-ui/core";

import "./style.scss";

import type {LoginFormStoreType} from "../../storeTypes/LoginForm";


const RenderError = ({errorMsg}: any) => (
  <div className="login__error">
    <div className="login__error--text">
      {errorMsg}
    </div>
    <div className="login__error--icon">
      {errorMsg &&
      <img
        alt="warn"
        src={warn}
      />
      }
    </div>
  </div>
);

type State = {
  passwordIsMasked: boolean
}


class LoginForm extends React.Component<LoginFormStoreType, State> {
  constructor (props: LoginFormStoreType) {
    super(props);
    this.state = {"passwordIsMasked": true};
    this.handleTogglePasswordMask = this.handleTogglePasswordMask.bind(this);
  }

  handleTogglePasswordMask: Function;

  handleTogglePasswordMask () {
    this.setState((prevState) => ({"passwordIsMasked": !prevState.passwordIsMasked}));
  }

  render () {
    const {passwordIsMasked} = this.state;
    const {onSubmit, status, errorMsg, isPristine, isInvalid} = this.props;
    const renderedError = status === "Error" && (<RenderError errorMsg={errorMsg} />);
    return (
      <div className="login">
        <div className="login__header">
          <img
            alt="logo"
            className="login__header--logo"
            src={logo}
          />
        </div>
        <form
          onSubmit={onSubmit}
        >
          <div
            className="login__text2"
          >{"Utilisez vos informations d’accès OCP pour accéder à l’application"}</div>
          <div className="login__email">
            <FormControl fullWidth>
              <Field
                component={TextField}
                id="login-email"
                label={"Identifiant/Email"}
                name="email"
              />
            </FormControl>
          </div>
          <div className="login__password">
            <FormControl fullWidth>
              <Field
                InputProps={{
                  "endAdornment": (
                    <InputAdornment
                      position="end"
                      style={{"marginLeft": "-25px"}}
                    >
                      <RemoveRedEye
                        onClick={this.handleTogglePasswordMask}
                      />
                    </InputAdornment>
                  )
                }}
                component={TextField}
                id="login-password"
                label={"Password"}
                name="password"
                type={passwordIsMasked
                  ? "password"
                  : "text"}
              />
            </FormControl>
          </div>
          {renderedError}
          <div className="login__button">
            <Field
              component={Button}
              disabled={isPristine || isInvalid}
              id="LoginForm-SubmitButton"
              name="submit"
              type="submit"
            >
              {"CONNEXION"}
            </Field>
          </div>
        </form>
      </div>
    );
  }
}

export default LoginForm;
