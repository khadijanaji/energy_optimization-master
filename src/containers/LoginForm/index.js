// @flow

import {reduxForm} from "redux-form";
import LoginForm from "../../components/LoginForm";
import {compose} from "redux";
import {connect} from "react-redux";
import {loginFormName, loginFormSelector} from "./selectors";
import {loginActionSuccessCreator} from "../../actions/loginActions";

const mapDispatchToProps = (dispatch) => ({
  "onSubmit": (event) => {
    event.preventDefault();
    dispatch(loginActionSuccessCreator({
      "email": "name",
      "name": "",
      "token": "",
      "roles": [
        "PAP28",
        "PAP54",
        "OSBL",
        "SAP",
        "DAP"
      ],
      "factories": [
        "JFC1",
        "JFC2",
        "JFC3",
        "JFC4"
      ]
    }));
    // @dispatch(push("/"));
  },
  dispatch
});

export default compose(connect(loginFormSelector, mapDispatchToProps), reduxForm({"form": loginFormName}))(LoginForm);
