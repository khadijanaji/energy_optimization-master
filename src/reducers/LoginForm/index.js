// @flow

import {
  LOGIN_ACTION,
  LOGIN_ACTION_ERROR,
  LOGIN_ACTION_SUCCESS,
  LOGOUT_ACTION
} from "../../actions/loginActions/constants";
import type {LoginFormStoreRecordType} from "../../storeTypes/LoginForm";
import {LoginFormStoreRecordFactory} from "../../storeTypes/LoginForm";
import LoginReducer from "./LoginReducer/LoginReducer";
import LoginSuccessReducer from "./LoginSuccessReducer/LoginSuccessReducer";
import LoginErrorReducer from "./LoginErrorReducer/LoginErrorReducer";
import LogoutReducer from "./LogoutReducer/LogoutReducer";

const DEFAULT_STATE = LoginFormStoreRecordFactory();

/**
 * Stores the dashboard information
 * @param {object} state The current state
 * @param {object} action The action to handle
 * @returns {object} The new state
 */
export default function LoginFormReducer (state: LoginFormStoreRecordType = DEFAULT_STATE, action: any) {
  switch (action.type) {
  case LOGIN_ACTION:
    return LoginReducer(state);
  case LOGIN_ACTION_SUCCESS:
    return LoginSuccessReducer(state, action);
  case LOGIN_ACTION_ERROR:
    return LoginErrorReducer(state, action);
  case LOGOUT_ACTION:
    return LogoutReducer(state);
  default:
    return state;
  }
}
