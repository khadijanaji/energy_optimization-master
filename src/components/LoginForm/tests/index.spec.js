/* globals describe it expect */

import {shallow} from "enzyme";
import React from "react";

import LoginForm from "..";

const propsWithError = {
  "onSubmit": () => ({}),
  "error": "Erreur",
  "isPristine": true,
  "isInvalid": true
};
const propsWithoutError = {
  "onSubmit": () => ({}),
  "error": null,
  "isPristine": false,
  "isInvalid": false
};

describe("Component", () => {
  describe("LoginForm Render", () => {
    it("With error Props: Should match snapshot", () => {
      const wrapper = shallow(<LoginForm {...propsWithError} />);
      expect(wrapper).toMatchSnapshot();
    });
    it("Without error Props: Should match snapshot", () => {
      const wrapper = shallow(<LoginForm {...propsWithoutError} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
