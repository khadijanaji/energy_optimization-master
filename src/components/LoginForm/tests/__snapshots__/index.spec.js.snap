// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoginForm Render With error Props: Should match snapshot 1`] = `
<div
  className="login"
>
  <div
    className="login__header"
  >
    <img
      alt="logo"
      className="login__header--logo"
      src=""
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login__text2"
    >
      Utilisez vos informations d’accès OCP pour accéder à l’application
    </div>
    <div
      className="login__email"
    >
      <WithStyles(FormControl)
        fullWidth={true}
      >
        <Field
          component={[Function]}
          id="login-email"
          label="Identifiant/Email"
          name="email"
        />
      </WithStyles(FormControl)>
    </div>
    <div
      className="login__password"
    >
      <WithStyles(FormControl)
        fullWidth={true}
      >
        <Field
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
                style={
                  Object {
                    "marginLeft": "-25px",
                  }
                }
              >
                <pure(RemoveRedEye)
                  onClick={[Function]}
                />
              </WithStyles(InputAdornment)>,
            }
          }
          component={[Function]}
          id="login-password"
          label="Password"
          name="password"
          type="password"
        />
      </WithStyles(FormControl)>
    </div>
    <div
      className="login__button"
    >
      <Field
        component={[Function]}
        disabled={true}
        id="LoginForm-SubmitButton"
        name="submit"
        type="submit"
      >
        CONNEXION
      </Field>
    </div>
  </form>
</div>
`;

exports[`Component LoginForm Render Without error Props: Should match snapshot 1`] = `
<div
  className="login"
>
  <div
    className="login__header"
  >
    <img
      alt="logo"
      className="login__header--logo"
      src=""
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login__text2"
    >
      Utilisez vos informations d’accès OCP pour accéder à l’application
    </div>
    <div
      className="login__email"
    >
      <WithStyles(FormControl)
        fullWidth={true}
      >
        <Field
          component={[Function]}
          id="login-email"
          label="Identifiant/Email"
          name="email"
        />
      </WithStyles(FormControl)>
    </div>
    <div
      className="login__password"
    >
      <WithStyles(FormControl)
        fullWidth={true}
      >
        <Field
          InputProps={
            Object {
              "endAdornment": <WithStyles(InputAdornment)
                position="end"
                style={
                  Object {
                    "marginLeft": "-25px",
                  }
                }
              >
                <pure(RemoveRedEye)
                  onClick={[Function]}
                />
              </WithStyles(InputAdornment)>,
            }
          }
          component={[Function]}
          id="login-password"
          label="Password"
          name="password"
          type="password"
        />
      </WithStyles(FormControl)>
    </div>
    <div
      className="login__button"
    >
      <Field
        component={[Function]}
        disabled={false}
        id="LoginForm-SubmitButton"
        name="submit"
        type="submit"
      >
        CONNEXION
      </Field>
    </div>
  </form>
</div>
`;
